packages/design-system/src/components/ButtonIcon/ButtonIcon.stories.mdx

```diff
@@ -0,0 +1,183 @@
+import { ArgsTable, Meta, Canvas, Story } from '@storybook/addon-docs';
+import { FigmaImage, FigmaLink, Links } from '../../docs';
+import ButtonIconPrimitive from './Primitive/ButtonIconPrimitive';
+import * as Stories from './ButtonIcon.stories';
+import InlineMessage from '../InlineMessage/index';
+
+<Meta title="Components/ButtonIcon" component={ButtonIconPrimitive} />
+
+<Links>
+	<FigmaLink href="https://www.figma.com/file/KuJ1PlP77uyXlfOhdniqsZ/ButtonIcon?node-id=1%3A75" />
+</Links>
+
+# ButtonIcon
+
+This component should be used when icons are meant to be clicked on.
+
+It handles the two largest usecases we have for clickable icons: actions and toggles.
+
+- Always try use when an icon is meant to be clicked on.
+- Always provide developers with copy for the tooltip.
+- If the button holds the ON / OFF state of something, then use the Toggle variants.
+- If the button is floating on top of something else, use the Floating variant.
+- Only use size XS sporadically.
+- Spacing between buttons should be at least 4px (XXS), 8px is best (XS).
+
+## Zoning
+
+<FigmaImage
+	src="https://www.figma.com/file/qc8oUSyVJM67gSsbTShftt/%F0%9F%93%90-Zoning?node-id=1049%3A415"
+	alt="zoning image for ButtonIcons"
+/>
+
+## Style
+
+### Variations
+
+<FigmaImage
+	src="https://www.figma.com/file/KuJ1PlP77uyXlfOhdniqsZ/ButtonIcon?node-id=3232%3A34023"
+	alt="variations image for ButtonIcon"
+/>
+
+#### Default
+
+This is the go-to ButtonIcon for most generic actions.
+
+```tsx
+import { ButtonIcon } from '@talend/design-system';
+
+function MyComponent() {
+	return (
+		<>
+			<ButtonIcon onClick={onAddDataset()}>Add a dataset</ButtonIcon>
```

Comment:
Is the purpose of the ButtonIcon is always to show a "plus" icon ?
I would think a ButtonIcon would lead also  to choose an icon

Reply:
Or to place an icon inside the component, but if we do so, why specifying a button "ButtonIcon" without puting an icon in it ?

Reply:
My bad, there would of course be an `icon` prop that takes a string that is necessarily one of the names present in the icon library. 
I forgot to put it in the example :/